### PR TITLE
7 user login

### LINKED ISSUE #7: [5pts] App Backend: User login
**Fill in the blanks**:
As a student, I would like to login to the app, so that I can sync/save my schedule and add other users as friends.

**Definition of Done - Check as you go!**
- [ ] Tasks have been implemented
- [ ] Documented
- [ ] Passing Tests for each task
- [ ] Pull Request made
- [ ] Code Reviewed

### PR DESCRIPTION
**Pull request by brendacano**
accidentally replaced the old branch, heres the new branch, its basically the same as what was in https://github.com/Bestagons/quack/pull/137

### CODE REVIEW

**Comment by brendacano** [comment]
If someone looks at this could you help me with the errors that are happening in the build
```
Traceback:
/opt/hostedtoolcache/Python/3.9.8/x64/lib/python3.9/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
server_test.py:1: in <module>
    import server
server.py:5: in <module>
    from auth_bearer import JWTBearer
auth_bearer.py:4: in <module>
    from auth_handler import decodeJWT
auth_handler.py:6: in <module>
    import jwt
E   ModuleNotFoundError: No module named 'jwt'
```
I know some of it is the testing too because of the response but I want to try to fix this first

**Comment by RafaelPiloto10** [comment]
Any luck with this: https://stackoverflow.com/questions/63309591/import-jwt-importerror-no-module-named-jwt

**Review by RafaelPiloto10 — APPROVED** [review]
Just fix the errors, but otherwise, lgtm

**Comment by brendacano** [comment]
> Just fix the errors, but otherwise, lgtm

Thanks for your feedback Rafael. I will try to fix the errors soon. I will most likely not get to it until Wednesday/Thursday just as an FYI. But I will try.

File: app_backend/.env.sample [hunk 1]
```diff
@@ -1,2 +1,5 @@
 DB_USERNAME="YOUR_USERNAME"
 DB_PASSWORD="YOUR_PASSWORD"
+
+SECRET="A SECRET KEY"
```
Comment (Foxworth22) [inline app_backend/.env.sample:4]: Is this an Easter egg?

File: app_backend/database.py [hunk 2]
```diff
@@ -39,3 +39,8 @@ def save_user_in_db(self, login_info: dict, dry_run = False):
             users.insert_one(login_info)
 
         return {"msg": "Successfully registered new user."}
```
Comment (Foxworth22) [inline app_backend/database.py:46]: This looks really nice so far! Great work Brenda!

File: app_backend/server_test.py [hunk 3]
```diff
@@ -75,7 +93,7 @@ def to_json(self):
         Params('test0000', 'test0@emory.edu', '', status.HTTP_400_BAD_REQUEST, {"err" : "Invalid password. It must be 8 to 20 characters and have at least one upper case letter, one lower case letter, and one digit."}),
         Params('Test. 500', 'test.emory.edu', 'Test0000', status.HTTP_400_BAD_REQUEST, {"err" : "Invalid email address. It must be an emory.edu email."}),
         Params('test0000', 'test0@emory.edu', 'TEST@123', status.HTTP_400_BAD_REQUEST, {"err" : "Invalid password. It must be 8 to 20 characters and have at least one upper case letter, one lower case letter, and one digit."}),
-         Params('User Test', 'test@emory.edu', 'Test0000', status.HTTP_400_BAD_REQUEST, {"err": "Email already exists. Use a different email."}),
```
Comment (Foxworth22) [inline app_backend/server_test.py]: What changed between this and line 96? Also why are there so many spaces between 78 & 96 or am I missing something?

**Review by Foxworth22 — APPROVED** [review]
Looks great to me! What does the time package do?

File: app_backend/.env.sample (hunk 1 above)
Comment (brendacano) [reply app_backend/.env.sample:4]: This is for the token stuff. It needs a secret key to encode it

File: app_backend/server_test.py (hunk 3 above)
Comment (brendacano) [reply app_backend/server_test.py]: I'm pretty sure it isn't showing those lines because they haven't changed from what was originally there

**Comment by RafaelPiloto10** [comment]
@brendacano @Foxworth22 please re-review

File: app_backend/tests/server_test.py [hunk 4]
```diff
@@ -28,13 +28,29 @@ def test_root():
 
 
 def test_login():
-    response = client.get("/user/login")
-    assert response.status_code == 200
-    assert response.json() == {"msg": "This is the login page!"}
```
Comment (Foxworth22) [inline app_backend/tests/server_test.py:33]: Just checking, this deletion was intentional, yes?

**Review by Foxworth22 — APPROVED** [review]
Looks like solid improvements. Good work!

File: app_backend/tests/server_test.py (hunk 4 above)
Comment (brendacano) [reply app_backend/tests/server_test.py:33]: This was the old thing that was there and it got replaced with a `client.post` so this was very much intentional.

File: app_backend/requirements.txt [hunk 5]
```diff
@@ -19,6 +19,7 @@ platformdirs==2.4.0
 pluggy==1.0.0
 py==1.10.0
 pydantic==1.8.2
+PyJWT==2.3.0
```
Comment (brendacano) [inline app_backend/requirements.txt:22]: perfect

**Review by brendacano — COMMENTED** [review]
The changes you made look good to me. Thank you for working on this 🙌😀

I think as long as it works and passes the tests and code coverage we are all set

**Pull request merged**